app/src/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -1796,9 +1843,25 @@ public void showView(View next, AnimationType from, boolean animateLastView) {
         mViewPane.addView(mCurrentView, lp);
 
         mCurrentView.startAnimation(mInAnimation);
-        if (mCurrentView instanceof ODKView)
+        
+        if (mCurrentView instanceof ODKView) {
             ((ODKView) mCurrentView).setFocus(this);
-        else {
+            
+            String groupLabelText = ((ODKView) mCurrentView).getGroupLabel();
```

Comment:
Whenever possible try to define a variable as close as possible to its usage (so down near line 1859). Unless it is a class level variable, which should go at the top of the class, contrary to what Clayton may say :p